src/menus/PluginMenus.cpp

```diff
@@ -144,231 +523,201 @@ bool CompareEffectsByType(const PluginDescriptor *a, const PluginDescriptor *b)
 
-   bool grouped = false;
-   if (groupBy.StartsWith(wxT("groupby")))
+   std::vector<MenuSectionBuilder> sections;
+   
+   auto MakeAddSortedItems = [=](SortBy sortby, bool useSubgroups)
    {
-      grouped = true;
-   }
-
-   // Some weird special case stuff just for Noise Reduction so that there is
-   // more informative help
-   const auto getBatchFlags = [&]( const PluginDescriptor *plug ){
-      if ( plug->GetSymbol().Msgid() == XO( "Noise Reduction" ) )
-         return
-            ( batchflags | NoiseReductionTimeSelectedFlag() ) & ~TimeSelectedFlag();
-      return batchflags;
+      return [=](MenuTable::BaseItemPtrs& items, std::vector<const PluginDescriptor*>& plugins)
+      {
+         return AddSortedEffectMenuItems(items, plugins, batchflags, realflags, sortby, useSubgroups);
+      };
    };
 
-   TranslatableStrings groupNames;
-   PluginIDs groupPlugs;
-   std::vector<CommandFlag> groupFlags;
-   if (grouped)
+   auto MakeAddGroupedItems = [=](GroupBy groupBy, bool useSubgroups)
    {
-      TranslatableString last;
-      TranslatableString current;
-
-      for (size_t i = 0; i < pluginCnt; i++)
+      return [=](MenuTable::BaseItemPtrs& items, std::vector<const PluginDescriptor*>& plugins)
       {
-         const PluginDescriptor *plug = plugs[i];
-
-         auto name = plug->GetSymbol().Msgid();
-
-         if (plug->IsEffectInteractive())
-            name += XO("...");
-
-         if (groupBy == wxT("groupby:publisher"))
-         {
-            current = EffectManager::Get().GetVendorName(plug->GetID());
-            if (current.empty())
-            {
-               current = XO("Unknown");
-            }
-         }
-         else if (groupBy == wxT("groupby:type"))
-         {
-            current = EffectManager::Get().GetEffectFamilyName(plug->GetID());
-            if (current.empty())
-            {
-               current = XO("Unknown");
-            }
-         }
-
-         if (current != last)
-         {
-            using namespace MenuTable;
-            BaseItemPtrs temp;
-            bool bInSubmenu = !last.empty() && (groupNames.size() > 1);
-
-            AddEffectMenuItemGroup(temp,
-               groupNames,
-               groupPlugs, groupFlags, isDefault);
-
-            table.push_back( MenuOrItems( wxEmptyString,
-               ( bInSubmenu ? last : TranslatableString{} ), std::move( temp )
-            ) );
-
-            groupNames.clear();
-            groupPlugs.clear();
-            groupFlags.clear();
-            last = current;
-         }
+         return AddGroupedEffectMenuItems(items, plugins, batchflags, realflags, groupBy, useSubgroups);
+      };
+   };
 
-         groupNames.push_back( name );
-         groupPlugs.push_back(plug->GetID());
-         groupFlags.push_back(
-            plug->IsEffectRealtime() ? realflags : getBatchFlags( plug ) );
+   auto DefaultFilter = [](auto plug) { return IsEnabledPlugin(plug) && IsDefaultPlugin(plug); };
```

Comment:
I see the new changes with `DefaultFilter` and how `IsDefaultPlugin` changed.

No objections.